consumers/writers/influxdb/consumer_test.go

```diff
@@ -88,22 +138,25 @@ func TestSaveSenml(t *testing.T) {
 	}
 
 	for _, tc := range cases {
-		// Clean previously saved messages.
-		_, err := queryDB(dropMsgs)
+		err := resetBucket()
 		require.Nil(t, err, fmt.Sprintf("Cleaning data from InfluxDB expected to succeed: %s.\n", err))
 
 		now := time.Now().UnixNano()
-		msg := senml.Message{
-			Channel:    "45",
-			Publisher:  "2580",
-			Protocol:   "http",
-			Name:       "test name",
-			Unit:       "km",
-			UpdateTime: 5456565466,
-		}
 		var msgs []senml.Message
 
+		chanID, err := idProvider.ID()
+		require.Nil(t, err, fmt.Sprintf("got unexpected error: %s", err))
+		pubID, err := idProvider.ID()
+		require.Nil(t, err, fmt.Sprintf("got unexpected error: %s", err))
 		for i := 0; i < tc.msgsNum; i++ {
+			msg := senml.Message{
+				Channel:    chanID,
+				Publisher:  pubID,
+				Protocol:   "http",
+				Name:       "test name",
+				Unit:       "km",
+				UpdateTime: 5456565466,
+			}
```

Comment:
Is there a reason to move this here?

Reply:
we do an attribute assignment inside the loop. when struct is created outside the loop, that attribute doesn't reset. 

Check the field function of consumer:

```go
func senmlFields(msg senml.Message) fields {
	ret := fields{
		"protocol":   msg.Protocol,
		"unit":       msg.Unit,
		"updateTime": msg.UpdateTime,
	}

	switch {
	case msg.Value != nil:
		ret["value"] = *msg.Value
	case msg.StringValue != nil:
		ret["stringValue"] = *msg.StringValue
	case msg.DataValue != nil:
		ret["dataValue"] = *msg.DataValue
	case msg.BoolValue != nil:
		ret["boolValue"] = *msg.BoolValue
	}

	if msg.Sum != nil {
		ret["sum"] = *msg.Sum
	}

	return ret
}
```

When we do not reset other fields of message, since `value` is the first field in switch case here, It always assings field as `value`. If Im not wrong, tests were checking only `value` field before.